### PR TITLE
sink(ticdc): fix broken test in cdc/sink/mq & migrate test-infra to testify

### LINKED ISSUE #2907: migrate test-infra to testify for `cdc/sink` pkg

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #2907

### What is changed and how it works?
1. fix goroutine leak in cdc/sink/mq/producer
2. migrate test-infra to testify & enable leak testing
3. fix broken mq_test

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/sink/mq/mq_flush_worker_test.go [hunk 1]
```diff
@@ -191,8 +194,12 @@ func TestBatch(t *testing.T) {
 
 			go func() {
 				for _, event := range test.events {
-					err := worker.addEvent(context.Background(), event)
-					require.NoError(t, err)
+					err := worker.addEvent(ctx, event)
+					if event.row == nil || event.row.CommitTs != math.MaxUint64 {
```
Comment (hi-rustin) [inline cdc/sink/mq/mq_flush_worker_test.go:198]: I don't quite understand this judgment, can you explain it?

File: cdc/sink/mq/mq_test.go [hunk 2]
```diff
@@ -70,16 +92,15 @@ func (s mqSinkSuite) TestKafkaSink(c *check.C) {
...
 	sink, err := NewKafkaSaramaSink(ctx, sinkURI, fr, replicaConfig, opts, errCh)
-	c.Assert(err, check.IsNil)
+	require.Nil(t, err)
 
 	encoder := sink.encoderBuilder.Build()
 
-	c.Assert(encoder, check.FitsTypeOf, &codec.JSONEventBatchEncoder{})
-	c.Assert(encoder.(*codec.JSONEventBatchEncoder).GetMaxBatchSize(), check.Equals, 1)
-	c.Assert(encoder.(*codec.JSONEventBatchEncoder).GetMaxMessageBytes(), check.Equals, 1048576)
+	// c.Assert(encoder, check.FitsTypeOf, &codec.JSONEventBatchEncoder{})
```
Comment (hi-rustin) [inline cdc/sink/mq/mq_test.go:99]: Why we need to comment this?

**Review by hi-rustin — COMMENTED** [review]
Thanks!

File: cdc/sink/mq/mq_test.go (hunk 2 above)
Comment (CharlesCheung96) [reply cdc/sink/mq/mq_test.go]: That is an oversight, thanks for your review.

File: cdc/sink/mq/mq_flush_worker_test.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sink/mq/mq_flush_worker_test.go:198]: Could you please add a comment here? Thanks!

**Review by hi-rustin — APPROVED** [review]
Thanks for working on this. Could you please resolve the conflict?

**Review by amyangfei — APPROVED** [review]

**Pull request merged**